lib/services/kubernetes.go

```diff
@@ -124,6 +124,10 @@ func MarshalKubeCluster(kubeCluster types.KubeCluster, opts ...MarshalOption) ([
 		return nil, trace.Wrap(err)
 	}
 
+	if c, ok := kubeCluster.(types.DiscoveredEKSCLuster); ok {
```

Comment:
What's the reasoning for marshaling the resource to json?

Reply:
It preexisting code that is used with operations on backend, I just fix it here, so it keeps working correctly.